adds Offline Icebreakers page

hannahkane opened the pull request:
Fixes #1881

**Changes proposed in this pull request:**
- create /pages/offline-icebreakers.jsx
- update ActivitiesPage.jsx to add new item for Offline Icebreakers and to *remove* LoFi NoFi item
- add Route for new page

hannahkane commented:
@mmmavis - R?

mmmavis commented:
Hey @hannahkane 😄 

in `less/components/sidebar.less`, around line 40 we should add `&.offline-icebreakers` there

```less
&.teaching-materials,
&.web-lit-basics,
&.web-lit-basics-two,
&.protect-your-data,
&.privacy-basics,
&.madewithcode-landing,
&.madewithcode-firstwebpage,
&.madewithcode-meme,
&.madewithcode-poster,
&.maker-party-2015,
&.parapara-page,
&.offline-icebreakers {
  .pageColorizer(@pinkSidebarColor);
}
```

This ensures the sidebar colour on the offline activity page is set to `@pinkSidebarColor`. Notice that we get the page class name `offline-icebreakers` from `pageClassName` in `offline-icebreakers.jsx` around line 106. 

#### What is `&.offline-icebreakers` ?####

The dot `.` is a CSS selector for `class`. See MDN for docs: https://developer.mozilla.org/en-US/docs/Web/CSS/Class_selectors

The ampersand `&` is a little tricky to explain. See http://blog.slaks.net/2013-09-29/less-css-secrets-of-the-ampersand/ for explanation. For this specific case you can think of it as selecting HTML elements that belong to class `page` AND (&) `offline-icebreakers`

Let me know if you need more clarification! 🍻

mmmavis commented:
everything else looks great @hannahkane 👯

hannahkane commented:
Thanks for the explanation, @mmmavis. Added.

Out of curiosity, why didn't I have to do this for other pages I've added, such as the Intermediate Web Literacy modules?

mmmavis commented:
> Out of curiosity, why didn't I have to do this for other pages I've added, such as the Intermediate Web Literacy modules?

that's cuz it's using the an existing class name, ie., `pageClassName: 'web-lit-basics'`

mmmavis commented on less/components/sidebar.less:
```diff
@@ -36,6 +36,9 @@ html.no-js .sidebar {
+  }
   &.parapara-page {
     .pageColorizer(@pinkSidebarColor);
   }
```
we should merge these two blocks since both share the same rule `.pageColorizer(@pinkSidebarColor);` 😉 

```less
&.teaching-materials,
&.web-lit-basics,
&.web-lit-basics-two,
&.protect-your-data,
&.privacy-basics,
&.madewithcode-landing,
&.madewithcode-firstwebpage,
&.madewithcode-meme,
&.madewithcode-poster,
&.maker-party-2015,
&.parapara-page,
&.offline-icebreakers {
  .pageColorizer(@pinkSidebarColor);
}
```

hannahkane commented:
whoops. sorry, @mmmavis. fixed now. thanks for your patience. :)

mmmavis commented:
ALL GOOD 👍 🚗 🚀
